solr/core/src/java/org/apache/solr/handler/component/QueryComponent.java

```diff
@@ -232,6 +233,47 @@ public void prepare(ResponseBuilder rb) throws IOException
     }
   }
 
+  // check if prefix is a prefix of the array
+  private static boolean isPrefix(SortField[] prefix, SortField[] array){
```

Comment:
There's an existing [ArrayUtil](https://github.com/apache/lucene-solr/blob/releases/lucene-solr/8.1.1/lucene/core/src/java/org/apache/lucene/util/ArrayUtil.java) class, I wonder if it might welcome a (generic) version of this method. What do you think?

Reply:
Given the fact that it is used only here I would keep it private - and we might put it into `ArrayUtil` when another class needs it.

Reply:
what do you think?

Reply:
Not sure. Yes, it's private and static but it seems a relatively long amount of what seems quite generic code and when another class needs something like it then it would be tricky to discover that the method already exists here and is available for relocation?

Having looked around a little, it looks like there's a [Collections.indexOfSubList](https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#indexOfSubList-java.util.List-java.util.List-) method in Java and if perhaps array-to-list conversion was acceptable then something like
```
0 == Collections.indexOfSubList(Arrays.asList(array), Arrays.asList(prefix))
```
might be another alternative?

Reply:
Ah! I think so, let me try!

Reply:
Done in 0563e66